components/compositing/compositor.rs

```diff
@@ -974,53 +1072,165 @@ impl<Window: WindowMethods + ?Sized> IOCompositor<Window> {
-                clip_id: ClipId::root(root_pipeline),
-            },
-            root_content_pipeline,
-            true,
-        );
+        let viewport_size = if cfg!(feature = "multiview") {
+            let dppx = self.page_zoom * self.hidpi_factor();
+            let viewport_size = self.embedder_coordinates.get_viewport().size.to_f32() / dppx;
+            let viewport_size = LayoutSize::from_untyped(viewport_size.to_untyped());
+            for (_, webview) in self.webviews.painting_order() {
+                if let Some(pipeline_id) = webview.pipeline_id {
+                    let rect = webview.rect / dppx;
+                    builder.push_iframe(
+                        LayoutRect::from_untyped(&rect.to_untyped()),
+                        LayoutRect::from_untyped(&rect.to_untyped()),
+                        &SpaceAndClipInfo {
+                            spatial_id: zoom_reference_frame,
+                            clip_id: ClipId::root(pipeline_id.to_webrender()),
+                        },
+                        pipeline_id.to_webrender(),
+                        true,
+                    );
+                }
+            }
+            viewport_size
+        } else {
+            let root_content_pipeline = self
+                .root_content_pipeline
+                .id
+                .expect("checked above")
+                .to_webrender();
+            let viewport_size = LayoutSize::new(
+                self.embedder_coordinates.get_viewport().width() as f32,
+                self.embedder_coordinates.get_viewport().height() as f32,
+            );
+            let viewport_rect = LayoutRect::new(LayoutPoint::zero(), viewport_size);
+            builder.push_iframe(
+                viewport_rect,
+                viewport_rect,
+                &SpaceAndClipInfo {
+                    spatial_id: zoom_reference_frame,
+                    clip_id: ClipId::root(root_pipeline),
+                },
+                root_content_pipeline,
+                true,
+            );
+            viewport_size
+        };
+
         let built_display_list = builder.finalize();
 
         // NB: We are always passing 0 as the epoch here, but this doesn't seem to
         // be an issue. WebRender will still update the scene and generate a new
         // frame even though the epoch hasn't changed.
+        let viewport_size = LayoutSize::from_untyped(viewport_size.to_untyped());
         transaction.set_display_list(
             WebRenderEpoch(0),
             None,
-            viewport_rect.size,
+            viewport_size,
             built_display_list,
             false,
         );
     }
 
-    fn set_frame_tree(&mut self, frame_tree: &SendableFrameTree) {
-        debug!(
-            "Setting the frame tree for pipeline {:?}",
-            frame_tree.pipeline.id
-        );
+    fn update_webview(&mut self, frame_tree: &SendableFrameTree) {
+        debug!("{}: Updating web view", frame_tree.pipeline.id);
```

Comment:
nit: in logging and comments, use “webview” (one word)

Reply:
fixed in ([`070398d`](https://github.com/servo/servo/pull/31417/commits/070398d0030a23be7ae9581d8a486ad473678ca6))

Reply:
thanks! there are still 17 other instances of “web view” in this patch, can you update those too?

Reply:
fixed in ([`ade8c27`](https://github.com/servo/servo/pull/31417/commits/ade8c27dfad50361a3c0d04567968cb297edc454))